### PR TITLE
[ST] UserST, improve checks and move testUOListeningOnlyUsersInSameCluster from KafkaST to UserST

### PR DESCRIPTION
**Pull request by henryZrncik**
### Type of change
- Refactoring

### Description

modification and relocation of `testUOListeningOnlyUsersInSameCluster`
- moved from `KafkaST` to `UserST` 
- add check of Secret labels when KafkaUser is watched by User Operator.  (Internal client being used)
- add check that Kafka user is really created as result of actions performed by relevant User Operator. (User with quota is used, in order for it to be listed with a usage of script `kafka-config.sh`, this in turn limits the test to work only with kraft disabled (more at https://issues.apache.org/jira/browse/KAFKA-13964, which actually still seems to be an issue), 

### Checklist

- [x] Write tests
- [x] Make sure all tests pass
- [x] Update documentation

fixes https://github.com/strimzi/strimzi-kafka-operator/issues/8260

### CODE REVIEW

File: systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java [hunk 1]
```diff
@@ -390,6 +396,97 @@ void testTlsExternalUser(ExtensionContext extensionContext) {
...
+            LOGGER.error(e);
+        }
+
+        LOGGER.info("Cleaning up Kafka: {}/{}", namespace, userListeningClusterName);
+        try {
+            KafkaResource.kafkaClient().inNamespace(namespace).withName(userListeningClusterName).withPropagationPolicy(DeletionPropagation.FOREGROUND).delete();
+            KafkaUtils.waitForKafkaDeletion(namespace, userListeningClusterName);
+        } catch (Exception e) {
+            LOGGER.error(e);
+        }
```
Comment (henryZrncik) [inline systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java:487]: i failed to notice if this is common practise here, but as I am working in shared namespace i wanted to safely clean after it (and as I am not interested in testing deletions it is in try/catch blocks) WDYT ?

File: systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java [hunk 2]
```diff
@@ -390,6 +396,97 @@ void testTlsExternalUser(ExtensionContext extensionContext) {
...
+     *     - Secret with label referencing Kafka Cluster B is created, and Kafka resource user corresponding to created KafkaUser custom resource exists only in Kafka Cluster B
+     *
+     * @usecase
+     *  - labels
+     *  - user-operator
+     *  - users
+     *  - secrets
+     */
+    @ParallelTest
+    @KRaftNotSupported("Probably bug in Kafka - https://issues.apache.org/jira/browse/KAFKA-13964")
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java:415]: The issue is resolved, could you give it a try in KRaft mode?

File: systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java (hunk 1 above)
Comment (Frawless) [reply systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java:487]: ResourceManager should handle it. If no, it is probably a bug.

File: systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java (hunk 2 above)
Comment (henryZrncik) [reply systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java]: true, this change (resolution) will actually take place from kafka version 3.5.0 onwards.

File: systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java (hunk 1 above)
Comment (henryZrncik) [reply systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java]: thank you, removed.

**Comment by henryZrncik** [comment]
@strimzi-ci run tests --testcase=UserST

File: systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java [hunk 3]
```diff
@@ -390,6 +394,79 @@ void testTlsExternalUser(ExtensionContext extensionContext) {
...
+
+        LOGGER.info("Verifying that Secret corresponding to KafkaUser: {}/{} has expected label: {}={}", namespace, testStorage.getUsername(), Labels.STRIMZI_CLUSTER_LABEL, userListeningClusterName);
+        TestUtils.waitFor("Wait for the Secret {}: {}/{} to exist while also having label to corresponding Kafka cluster", Constants.GLOBAL_POLL_INTERVAL, Constants.GLOBAL_TIMEOUT,
+            () -> kubeClient().listSecrets(namespace)
+                    .stream()
+                    .anyMatch(secret -> testStorage.getUsername().equals(secret.getMetadata().getName())
+                        && secret.getMetadata().getLabels().containsKey(Labels.STRIMZI_CLUSTER_LABEL)
+                        && secret.getMetadata().getLabels().get(Labels.STRIMZI_CLUSTER_LABEL).equals(userListeningClusterName)
+                    )
+        );
```
Comment (see-quick) [inline systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java:456]: Can we encapsulate such logic into `*utils` classes? :)

File: systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java [hunk 4]
```diff
@@ -390,6 +394,79 @@ void testTlsExternalUser(ExtensionContext extensionContext) {
...
+
+        LOGGER.info("Verifying KafkaUser: {}/{} is mapped into Kafka resource user in Kafka: {}/{}", namespace, testStorage.getUsername(), namespace, userListeningClusterName);
+        TestUtils.waitFor("Wait for the KafkaUser CR mapping into a Kafka user resource", Constants.GLOBAL_POLL_INTERVAL, Constants.GLOBAL_TIMEOUT,
+            () -> {
+                String getUserResult = KafkaCmdClient.describeUserUsingPodCli(namespace, scraperPodName, KafkaResources.plainBootstrapAddress(userListeningClusterName), "CN=" + testStorage.getUsername());
+                return getUserResult.contains(testStorage.getUsername());
+            });
```
Comment (see-quick) [inline systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java:463]: Same with this one. Generally in our test cases, we tried to minimize such code. Could this be also somehow refactored into utils classes?

**Review by see-quick — APPROVED** [review]
Just one nit but anyway looks good. Thanks

**Review by Frawless — APPROVED** [review]

**Pull request merged**